OMCompiler/Compiler/BackEnd/BackendDAEOptimize.mo

```diff
@@ -1495,6 +1494,26 @@ algorithm
       else ();
     end match;
   end for;
+
+  // Collect function if there is a data recon structure.
+  () := match inShared.dataReconciliationData
+    case NONE() then ();
+
+    case SOME(BackendDAE.DATA_RECON(symbolicJacobian =
+                      BackendDAE.GENERIC_JACOBIAN(jacobian=SOME((bdae, _, _, _, _, _)))))
+      algorithm
+        bdae := BackendDAEUtil.setFunctionTree(bdae, inFunctions);
+        shared := bdae.shared;
+        usedfuncs := removeUnusedFunctions(bdae.eqs, shared, {}, shared.functionTree, inUsedFunctions);
+        outUsedFunctions := DAE.AvlTreePathFunction.join(outUsedFunctions, usedfuncs);
+      then ();
+
+    // If data isSome and we have non generic jacobian what do we do?
```

Comment:
That cannot happen, data recon only creates a generic jacobian since it uses the `createJacobian` function.